File: Graph: Depth-First Search/ProgramTest.cpp
// This file is initialized with a code version of this
// question's sample test case. Feel free to add, edit,
// or remove test cases in this file as you see fit!

class ProgramTest : public TestSuite {
public:
  void Run() {
    RunTest("Test Case 1", []() {
      Node graph("A");
      graph.addChild("B")->addChild("C")->addChild("D");
      graph.children[0]->addChild("E")->addChild("F");
      graph.children[2]->addChild("G")->addChild("H");
      graph.children[0]->children[1]->addChild("I")->addChild("J");
      graph.children[2]->children[0]->addChild("K");

      vector<string> expected{"A", "B", "E", "F", "I", "J",
                              "C", "D", "G", "K", "H"};
      vector<string> inputArray{};
      assert(graph.depthFirstSearch(&inputArray) == expected);
    });
  }
};

File: Graph: Depth-First Search/ProgramTest.java
// This file is initialized with a code version of this
// question's sample test case. Feel free to add, edit,
// or remove test cases in this file as you see fit!

import java.util.*;

class ProgramTest {
  @Test
  public void TestCase1() {
    Program.Node graph = new Program.Node("A");
    graph.addChild("B").addChild("C").addChild("D");
    graph.children.get(0).addChild("E").addChild("F");
    graph.children.get(2).addChild("G").addChild("H");
    graph.children.get(0).children.get(1).addChild("I").addChild("J");
    graph.children.get(2).children.get(0).addChild("K");
    String[] expected = {"A", "B", "E", "F", "I", "J", "C", "D", "G", "K", "H"};
    List<String> inputArray = new ArrayList<String>();
    Utils.assertTrue(compare(graph.depthFirstSearch(inputArray), expected));
  }

  public static boolean compare(List<String> arr1, String[] arr2) {
    if (arr1.size() != arr2.length) {
      return false;
    }
    for (int i = 0; i < arr1.size(); i++) {
      if (!arr1.get(i).equals(arr2[i])) {
        return false;
      }
    }
    return true;
  }
}
